File: code_samples/dart/functional_monads.dart
// Functional: Monads
// AI/ML Training Sample

class Monads {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Monads();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/scala/functional_monads.scala
/**
 * Functional: Monads
 * AI/ML Training Sample
 */

class Monads {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object MonadsExample {
  def main(args: Array[String]): Unit = {
    val instance = new Monads()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
